DigitalLearningSolutions.Web/Controllers/Signposting/ViewResourceController.cs

```diff
@@ -0,0 +1,58 @@
+﻿namespace DigitalLearningSolutions.Web.Controllers.Signposting
+{
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Helpers;
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.FeatureManagement.Mvc;
+
+    [FeatureGate(FeatureFlags.UseSignposting)]
+    [Authorize(Policy = CustomPolicies.UserOnly)]
+    public class ViewResourceController : Controller
+    {
+        private readonly ILearningHubAuthApiUrlService learningHubAuthApiUrlService;
+        private readonly ILearningResourceReferenceDataService learningResourceReferenceDataService;
+
+        private readonly IUserService userService;
+
+        public ViewResourceController(
+            IUserService userService,
+            ILearningResourceReferenceDataService learningResourceReferenceDataService,
+            ILearningHubAuthApiUrlService learningHubAuthApiUrlService
+        )
+        {
+            this.userService = userService;
+            this.learningResourceReferenceDataService = learningResourceReferenceDataService;
+            this.learningHubAuthApiUrlService = learningHubAuthApiUrlService;
+        }
+
+        [Route("Signposting/ViewResource/{resourceReferenceId}")]
+        public IActionResult Index(int resourceReferenceId)
+        {
+            var delegateId = User.GetCandidateIdKnownNotNull();
+            var learningHubAuthId = userService.GetDelegateUserLearningHubAuthId(delegateId);
+
+            if (learningHubAuthId.HasValue)

```

Comment:
I would be tempted tp reverse this, checking for `if (!learningHubAuthId.HasValue)` then this larger body of code would not need to be nested.